spec-0010/index.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "SPEC 10 — Changelog"
+number: 10
+date: 2024-06-03
+author:
+  - "Inessa Pawson <inessapawson@gmail.com>"
+discussion: https://discuss.scientific-python.org/t/
+endorsed-by:
+---
+
+## Description
+
+SPEC 10 provides guidelines and best practices for maintaining a changelog file
+for libraries in the Scientific Python ecosystem.
+
+### Core Project Endorsement
+
+<!--
+Briefly discuss what it means for a core project to endorse this SPEC.
+-->
+
+### Ecosystem Adoption
+
+The endorsement of this SPEC signifies your project's support for the guidelines 
+laid out in the document.
+
+## Implementation
+
+Keeping a clear, organized, and human-readable record of notable changes for each 
+version of a Python library is essential for the project’s maintenance and sustainability. 
+It promotes transparency and trust between the project’s leadership team, developers, and users.
+
+### What Is a Changelog?
+
+A changelog is a file which contains a curated, chronologically ordered list of 
+notable changes for each version of a project.
+
+### Guiding Principles
+
+- Changelogs are meant for humans, not machines.
+- Every version should have an entry.
+- Group similar types of changes together.
+- Make versions and sections linkable.
+- List the most recent version first.
+- Display the release date for each version.
+- Indicate if you follow Semantic Versioning.
+
+### Types of Changes
+
+- **Added**: New features.
+- **Changed**: Modifications in existing functionality.
+- **Deprecated**: Soon-to-be-removed features.
+- **Removed**: Features that are now removed.
+- **Fixed**: Bug fixes.
+- **Security**: Vulnerability fixes.
+
+### Standard Changelog Format
+There is not one standard changelog format.
+
+ADD EXAMPLES
+
+### Naming the Changelog File
+CHANGELOG, HISTORY, NEWS, or RELEASES are widely accepted names for changelog files.
+
+### GitHub Releases
+
+TO DO
+
+### Automation
+
+TO DO
```

Comment:
I would recommend adding a link to the PyPI instructions for automation here: https://packaging.python.org/en/latest/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/

Reply:
but this document is not about releasing, but just the changelog, and thus automation for me means e.g. the github actions checking the existence of a changelog, or the changelog framework itself, e.g. creating one single changelog file from the fragments for towncrier

Reply:
In my mind, generating the changelog and the release are tightly coupled and usually happens at the same time, so for any automation to happen it would be in the same file. here is an example of an automated release pipeline generated with the above docs that also generates release notes automatically: https://github.com/duckdb/dbt-duckdb/blob/master/.github/workflows/release.yml

There is also a `--notes` tag that lets you add custom (generated or manually entered) notes

Reply:
I like to have the changelog available, and complete/correct during the dev process already, it's quite useful for big projects, but I also find it somewhat useful to enforce the presence the changelog at the time a contribution is made